lib/sli/unittest.sli

```diff
@@ -47,9 +47,19 @@ is_threaded_with_openmp not { exit_test_gracefully } if
 */
 /exit_test_gracefully
 {
-  statusdict/exitcodes/success :: quit_i
+  statusdict/exitcodes/skipped :: quit_i
 } def
 
+/exit_test_gracefully_reason[/stringtype]
+{
+  dup (no_mpi) eq {statusdict/exitcodes/skipped_no_mpi :: quit_i} if
+  dup (have_mpi) eq {statusdict/exitcodes/skipped_have_mpi :: quit_i} if
+  dup (no_openmp) eq {statusdict/exitcodes/skipped_no_openmp :: quit_i} if
+  dup (no_gsl) eq {statusdict/exitcodes/skipped_no_gsl :: quit_i} if
+  dup (no_music) eq {statusdict/exitcodes/skipped_no_music :: quit_i} if
+  statusdict/exitcodes/scripterror :: quit_i
+} bind def
+
```

Comment:
I would also change all "openmp" into "no_threads" or "not_threaded". OpenMP is an implementation detail.